tests/rptest/tests/control_character_flag_test.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import time
+
+import requests.exceptions
+from ducktape.mark import parametrize
+from ducktape.utils.util import wait_until
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST
+from rptest.services.redpanda_installer import RedpandaInstaller, wait_for_num_versions, ver_string
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class ControlCharacterPermittedBase(RedpandaTest):
+    feature_legacy_permit_control_char = 'legacy_permit_unsafe_log_operation'
+
+    def __init__(self, test_context, **kwargs):
+        super(ControlCharacterPermittedBase,
+              self).__init__(test_context=test_context, **kwargs)
+        self._installer = self.redpanda._installer
+
+    def setUp(self):
+        # handled by test case to support parameterization
+        pass
+
+    def _start_redpanda(self):
+        super(ControlCharacterPermittedBase, self).setUp()
+
+    def _validate_pre_upgrade(self, version):
+        self._installer.install(self.redpanda.nodes, version)
+        super(ControlCharacterPermittedBase, self).setUp()
+
+        unique_versions = wait_for_num_versions(self.redpanda, 1)
+        assert ver_string(version) in unique_versions
+
+        admin = Admin(self.redpanda)
```

Comment:
Try putting this in the constructor or something like that since you create `Admin` in multiple places in this test file.